CHANGELOG.md

```diff
@@ -1,7 +1,13 @@
 ## vNext (TBD)
 
+### Breaking Changes
+* The constructor of `ClientResetError` is changed to require error category and error code arguments and the `Configuration` parameter is now replaced by the `App` parameter. The `code` that was of type `SyncClientErrorCode` is now of type `SyncSessionErrorCode`. In case of fallback on client reset `isAutomaticClientReset` will return true. `SyncClientErrorCode.autoClientResetFailure` is no more returned as a code of `ClientResetError` ([#1291](https://github.com/realm/realm-dart/pull/1291)).
```

Comment:
I am not sure why do we need to introduce these breaking changes

Reply:
I added a description on the top of this PR. This is how the .Net SDK returns the errors, they are not Client errors, but Session errors. The difference here is the new property `isAutomaticClientReset` that have added to have a way the users to distinguish the client recovery fallback.

Reply:
If the users have attached `ManualRecoveryHandler` and they want to handle the error themselves, we have to provide to them the correct `SyncSessionErrorCode`, reported during the sync process.
If the users have attached `RecoverUnsyncedChangesHandler`, `RecoverOrDiscardUnsyncedChangesHandler` or `DiscardUnsyncedChangesHandler` we handle the client recovery for them, but if it fails the `ClientReserError` is returned  
to the `ManualResetFallback` and then the error code is `SyncClientErrorCode.autoClientResetFailure`, because the error is on the client. Since we have different enums for the errors `SyncClientErrorCode` and `SyncSessionErrorCode` we can not return the client error from the same property `code`, which is of type `SyncSessionErrorCode`. This is one of the ways to handle this issue. The other option is just to add a new property `sessionCode` for the session errors, and to set the existing property for the client error to `Unknown` in case of a session error. In both cases we will introduce some breaking changes.

Reply:
Ok, let me think if we can make it without breaking changes with only adding new properties and named parameters to the constructor.